boot/zephyr/flash_map.c

```diff
@@ -355,6 +355,6 @@ int flash_area_get_sectors(int idx, uint32_t *cnt, struct flash_sector *ret)
 {
     struct layout_data data;
 
-    return flash_area_layout(idx, cnt, ret, get_sectors_cb, &data);
+    return flash_area_layout(idx, (int*)cnt, ret, get_sectors_cb, &data);
```

Comment:
Can you add the cast in flash_area_to_sectors(), and change the prototype of flash_area_layout()? That would be cleaner since flash_area_to_sectors() is the "old style" routine, and struct layout_data already wants an unsinged value to assign to ret_len and pull from ret_idx.